### PR TITLE
Fixes to support Generating API Documentation

### PR DESCRIPTION
**Pull request by audionerd**
Companion PR for https://github.com/statelyai/docs/pull/237, which now uses `xstate` as a submodule, and requires these changes in order to build API docs.

This PR fixes issues that prevented the XState API docs index page from rendering properly, and fixes two warnings that occurred when API Documenter ran.

Includes and closes https://github.com/statelyai/xstate/pull/4374

### CODE REVIEW

**Review by davidkpiano — APPROVED** [review]

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -284,6 +287,8 @@ export interface StateValueMap {
 /**
  * The string or object representing the state value relative to the parent state node.
  *
+ * @remarks
```
Comment (Andarist) [inline packages/core/src/types.ts:290]: I'm still quite unsure why we need this. RxJS has *extensive* and long JSDocs and they use api-extractor too but there is no `@remarks` in sight in their codebase. I wonder what's the difference? 🤔

Comment (audionerd) [reply packages/core/src/types.ts:290]: On the index page a description is provided for each item.

With `@remarks`, we can choose to show only the first summary line in the descriptions:
index-with-remarks

Without `@remarks`, the description is longer (and most of the index descriptions are a single line so this one stands out):
index-without-remarks

So it’s a subjective change, but I thought it was better to have a shorter description. If you disagree I’m happy to change it back.

I had also noticed the list items weren’t formatted properly on the index, so I wanted to omit them, but I just found a fix for that I can push to this PR, so it's not a deciding factor.

Comment (audionerd) [reply packages/core/src/types.ts:290]: API Extractor uses TSDoc, and `@remarks` is specific to TSDoc. See: https://tsdoc.org/pages/tags/remarks

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -835,7 +841,7 @@ export interface StateNodeConfig<
     TDelay
   >;
   /**
-   * @private
+   * @internal
```
Comment (Andarist) [inline packages/core/src/types.ts:844]: We are using `stripInternal` and I'm not sure if we want to "hide" this parent property like this. I'd remove this particular comment completely.

**Review by Andarist — APPROVED** [review]

File: packages/core/src/types.ts (hunk 1 above)
Comment (audionerd) [reply packages/core/src/types.ts:290]: That's a good point. I’ll see what I can find out about TSDocs in VS Code. Might be some ways to improve the experience.

**Pull request merged**